### PR TITLE
update export, req_project, rsync cmds

### PR DESCRIPTION
**Pull request by juanvallejo**
cc @soltysh

### CODE REVIEW

File: pkg/oc/cli/cmd/export.go [hunk 1]
```diff
@@ -62,64 +92,79 @@ func NewCmdExport(fullName string, f *clientcmd.Factory, streams genericclioptio
...
+	cmd.Flags().BoolVar(&o.Raw, "raw", o.Raw, "If true, do not alter the resources in any way after they are loaded.")
+	cmd.Flags().StringVarP(&o.Selector, "selector", "l", o.Selector, "Selector (label query) to filter on")
+	cmd.Flags().BoolVar(&o.AllNamespaces, "all-namespaces", o.AllNamespaces, "If true, list the requested object(s) across all namespaces. Namespace in current context is ignored even if specified with --namespace.")
+	cmd.Flags().StringSliceVarP(&o.Filenames, "filename", "f", o.Filenames, "Filename, directory, or URL to file for the resource to export.")
 	cmd.MarkFlagFilename("filename")
 	cmd.Flags().Bool("all", true, "DEPRECATED: all is ignored, specifying a resource without a name selects all the instances of that resource")
 	cmd.Flags().MarkDeprecated("all", "all is ignored because specifying a resource without a name selects all the instances of that resource")
-	kcmdutil.AddPrinterFlags(cmd)
+	cmd.Flags().StringVar(&o.OutputVersion, "output-version", o.OutputVersion, "The preferred API versions of the output objects")
```
Comment (juanvallejo) [inline pkg/oc/cli/cmd/export.go]: @soltysh ideally this whole command would be deprecated in the future - so I don't think we should worry about not supporting output-version specifically

File: pkg/oc/cli/cmd/export.go [hunk 2]
```diff
@@ -62,64 +92,79 @@ func NewCmdExport(fullName string, f *clientcmd.Factory, streams genericclioptio
...
-	cmd.Flags().StringSliceVarP(&filenames, "filename", "f", filenames, "Filename, directory, or URL to file for the resource to export.")
+	o.PrintFlags.AddFlags(cmd)
+
+	cmd.Flags().StringVar(&o.AsTemplateName, "as-template", o.AsTemplateName, "Output a Template object with specified name instead of a List or single object.")
+	cmd.Flags().BoolVar(&o.Exact, "exact", o.Exact, "If true, preserve fields that may be cluster specific, such as service clusterIPs or generated names")
+	cmd.Flags().BoolVar(&o.Raw, "raw", o.Raw, "If true, do not alter the resources in any way after they are loaded.")
+	cmd.Flags().StringVarP(&o.Selector, "selector", "l", o.Selector, "Selector (label query) to filter on")
+	cmd.Flags().BoolVar(&o.AllNamespaces, "all-namespaces", o.AllNamespaces, "If true, list the requested object(s) across all namespaces. Namespace in current context is ignored even if specified with --namespace.")
+	cmd.Flags().StringSliceVarP(&o.Filenames, "filename", "f", o.Filenames, "Filename, directory, or URL to file for the resource to export.")
```
Comment (soltysh) [inline pkg/oc/cli/cmd/export.go]: You should wire filenames differently.

File: pkg/oc/cli/cmd/export.go (hunk 1 above)
Comment (soltysh) [reply pkg/oc/cli/cmd/export.go]: Yeah, but that'd also mean we're breaking backwards compatibility and I don't think we want to do that.

File: pkg/oc/cli/cmd/export.go [hunk 3]
```diff
@@ -62,64 +92,79 @@ func NewCmdExport(fullName string, f *clientcmd.Factory, streams genericclioptio
...
 
+	o.PrintObj = printer.PrintObj
+
+	o.Namespace, o.RequireNamespace, err = f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+
+	o.Builder = f.NewBuilder()
+	o.Builder.ResourceTypeOrNameArgs(true, args...)
```
Comment (soltysh) [inline pkg/oc/cli/cmd/export.go]: pass `args` to `ExportOptions`.

File: pkg/oc/cli/cmd/export.go [hunk 4]
```diff
@@ -62,64 +92,79 @@ func NewCmdExport(fullName string, f *clientcmd.Factory, streams genericclioptio
...
+
+	_, o.Typer = f.Object()
+	return nil
+}
+
+func (o *ExportOptions) Validate(cmd *cobra.Command) error {
+	if o.Exact && o.Raw {
+		return kcmdutil.UsageErrorf(cmd, "--exact and --raw may not both be specified")
```
Comment (soltysh) [inline pkg/oc/cli/cmd/export.go]: Try not to pass `cmd` here

**Comment by juanvallejo** [comment]
@soltysh thanks, review comments addressed

**Pull request merged**